src/secrets/CommandRemove.ts

```diff
@@ -20,9 +21,17 @@ class CommandRemove extends CommandPolykey {
     this.addOption(binOptions.clientPort);
     this.addOption(binOptions.recursive);
     this.action(async (secretPaths, options) => {
-      secretPaths = secretPaths.map((path: string) =>
-        binParsers.parseSecretPathValue(path),
-      );
+      for (let i = 0; i < secretPaths.length; i++) {
+        const value: string = secretPaths[i];
+        const parsedValue = binParsers.parseSecretPath(value);
+        // The vault root cannot be deleted
+        if (parsedValue[1] == null) {
+          throw new errors.ErrorPolykeyCLIRemoveSecret(
+            'EPERM: Cannot remove vault root',
+          );
+        }
```

Comment:
I am doing this here manually, as the generated errors aren't really descriptive. This needs to be changed in Polykey, but is not immediately required for this PR.

Reply:
This is actually being tracked in https://github.com/MatrixAI/Polykey/issues/826